worker/download.go

```diff
@@ -75,10 +76,9 @@ type (
 		numOverpaid    uint64
 		numRelaunched  uint64
 
-		unusedHostSectors map[types.PublicKey][]sectorInfo
-
-		sectors [][]byte
-		errs    HostErrorSet
+		pending   []*sectorDownloadInfo
```

Comment:
@ChrisSchinnerl I'm proposing to essentially inverse it and have a list of sectors in which we track the number of times it was selected for downloading. In order to keep the algorithm simple and the refactor to a minimum, I wanted to select from the sectors depending on how many times they've been selected already. Favouring indices that haven't been selected yet.

Reply:
Yeah I think keeping the number of times we have tried a sector and sorting by it instead of actually tracking whether it's being downloaded at the moment is a lot nicer and less likely to cause errors.